### PR TITLE
Persist auth response from tesla to redis and write tests for new function

### PR DESCRIPTION
**Pull request by 0xdev22**
# Proposed Changes

### CODE REVIEW

File: internal/controllers/user_integrations_auth_controller.go [hunk 1]
```diff
@@ -73,8 +90,18 @@ type DeviceDefinition struct {
 // @Security    BearerAuth
 // @Router      /integration/:tokenID/credentials [post]
 func (u *UserIntegrationAuthController) CompleteOAuthExchange(c *fiber.Ctx) error {
-	tokenID := c.Params("tokenID")
+	//Get the current user
```
Comment (elffjs) [inline internal/controllers/user_integrations_auth_controller.go:93]:
```suggestion
	// Get the current user
```

**Comment by elffjs** [comment]
I agree with the linter—we're not passing enough arguments to the controller constructor anymore.

**Review by elffjs — CHANGES_REQUESTED** [review]
I agree with the linter—we're not passing enough arguments to the controller constructor anymore.

**Comment by elffjs** [comment]
I think we must patch up some of the tests from last time to account for the new stuff in the handler. They're failing locally for me.

**Comment by elffjs** [comment]
(Other than that, the PR looks good)

**Review by elffjs — APPROVED** [review]

File: internal/controllers/user_integrations_auth_controller.go [hunk 2]
```diff
@@ -1,13 +1,21 @@
 package controllers
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
```
Comment (KevinJoiner) [inline internal/controllers/user_integrations_auth_controller.go:6]: nit: Should separate the std lib packages from other packages with a newline 
https://go.dev/wiki/CodeReviewComments#imports

**Review by KevinJoiner — APPROVED** [review]
One nit but LGTM

File: internal/controllers/user_integrations_auth_controller.go (hunk 2 above)
Comment (elffjs) [reply internal/controllers/user_integrations_auth_controller.go:6]: Is there a linter rule for this?

I think VS Code does this by default (whatever they're running on save) but I think Doyin uses GoLand and it's doing something else./

**Pull request merged**